### PR TITLE
General: Run applications as separate processes under linux

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Launched application on linux are child processes of OpenPype which may cause a huge issues of hanging processes, closing all children processes etc.

## Description
The goal is to create detached process on linux with creating mid-process that will launch the process as background process and exits. The mid process is new executable in OpenPype called `linux_app_launcher` which should work from code too.

The mid executable gives ability to set environment variables and detach the process.

## Notes
- it should be possible to detach the process in much easier way but I didn't find any that actually works

## How to test
1. Run application on linux
2. Check if the launched application is children process of OpenPype (using `pstree`)

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-2088 Run applications as separate processes under Linux](https://app.clickup.com/t/6658547/OP-2088)

**Comment by iLLiCiTiT** [comment]
I though about the environment variables and I think that they should be maybe set inside the mid process and not passed through `subprocess.Popen`. The reason is that it is possible to change `PYTHONPATH` or `PATH` or any `LD_LIBRARY_PATH` or any other variable which may be crutial for running the mid process.

**Review by 64qam — APPROVED** [review]

**Review by 64qam — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -103,6 +109,15 @@
     Executable("start.py", base=None,
                target_name="openpype_console", icon=icon_path.as_posix())
 ]
+if IS_LINUX:
+    executables.append(
+        Executable(
+            "linux_app_launcher.py",
```
Comment (antirotor) [inline setup.py:115]: I would call it just `app_launcher` as it will be generated only on linux boxes so it's little bit overhead

**Comment by antirotor** [comment]
I haven't tested it yet on machine it caused trouble with hanging processes, but I think this is good approach. There are some downfalls of using `&` that needs to be clear. In some cases (when it is used under shell) it will terminate the process whenever the shell is terminated and it all depends on shell configuration. So in theory it could cause Maya to be killed whenever you close shell you used to run OpenPype.

**Comment by iLLiCiTiT** [comment]
> So in theory it could cause Maya to be killed whenever you close shell you used to run OpenPype.

I think that opened Maya does not have any connection to OpenPype. Maya's parent process on launch is `app_launcher` (`linux_app_launcher`) which exits immediately.

**Pull request merged**